src/main/java/book/store/onlinebookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,75 @@
+package book.store.onlinebookstore.controller;
+
+import book.store.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import book.store.onlinebookstore.dto.cartitem.UpdateCartItemRequestDto;
+import book.store.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import book.store.onlinebookstore.model.User;
+import book.store.onlinebookstore.service.ShoppingCartService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Positive;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Validated
+@RequiredArgsConstructor
+@RequestMapping("/api/cart")
+@Tag(name = "Shopping cart management", description = "Endpoints for managing user shopping cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    @PreAuthorize("hasRole('ROLE_USER')")

```

Comment:
```suggestion
    @PreAuthorize("hasRole('USER')")
```
you can avoid using this prefix.